app/services/user_service.rb

```diff
@@ -4,9 +4,27 @@ def self.delete_spam(user)
     user.destroy
   end
 
+  def self.deactivate(user:, actor:, params:)
+    actor.ability.authorize! :deactivate, user
+    UserDeactivationResponse.create(user: user, body: params[:deactivation_response])
+    user.deactivate!
+    user
+  end
+
   def self.update(user:, actor:, params:)
     actor.ability.authorize! :update, user
-    user.update! params
+    user.update params
+    user
+  end
+
+  def self.change_password(user:, actor:, params:)
+    actor.ability.authorize! :update, user
+    user.errors.add :current_password, I18n.t(:"error.current_password_did_not_match") unless user.valid_password? params.delete(:current_password)
```

Comment:
Yeah; I've been having a good time recently with APIs using controller specs as defacto e2es... I test drive pretty much all of them out since it's usually pretty obvious what a new endpoint is supposed to do before it gets written.